internal/service/datazone/environment_profile.go

```diff
@@ -0,0 +1,361 @@
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+}
+
+func (r *resourceEnvironmentProfile) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	conn := r.Meta().DataZoneClient(ctx)
+
+	var plan, state dEnvProfile
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	if plan.EnvironmentBlueprintIdentifier.Equal(state.EnvironmentBlueprintIdentifier) &&
+		plan.ProjectIdentifier.Equal(state.ProjectIdentifier) && plan.DomainIdentifier.Equal(state.DomainIdentifier) {
+		in := &datazone.UpdateEnvironmentProfileInput{}
+
+		resp.Diagnostics.Append(flex.Expand(ctx, plan, in)...)
+		if resp.Diagnostics.HasError() {
+			return
+		}
+		in.Identifier = state.Id.ValueStringPointer()
+		out, err := conn.UpdateEnvironmentProfile(ctx, in)
+		if err != nil {
+			resp.Diagnostics.AddError(
+				create.ProblemStandardMessage(names.DataZone, create.ErrActionSetting, ResNameEnvironmentProfile, state.Id.ValueString(), err),
+				err.Error(),
+			)
+			return
+		}
+		state.AwsAccountId = flex.StringToFramework(ctx, out.AwsAccountId)
+		state.AwsAccountRegion = flex.StringToFramework(ctx, out.AwsAccountRegion)
+		state.CreatedAt = flex.TimeToFramework(ctx, out.CreatedAt)
+		state.CreatedBy = flex.StringToFramework(ctx, out.CreatedBy)
+		state.Description = flex.StringToFramework(ctx, out.Description)
+		state.DomainIdentifier = flex.StringToFramework(ctx, out.DomainId)
+		state.EnvironmentBlueprintIdentifier = flex.StringToFramework(ctx, out.EnvironmentBlueprintId)
+		state.Id = flex.StringToFramework(ctx, out.Id)
+		state.Name = flex.StringToFramework(ctx, out.Name)
+		state.ProjectIdentifier = flex.StringToFramework(ctx, out.ProjectId)
+		state.UpdatedAt = flex.TimeToFramework(ctx, out.UpdatedAt)
+
+		if resp.Diagnostics.HasError() {
+			return
+		}
+	}
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+}
+
+func (r *resourceEnvironmentProfile) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	conn := r.Meta().DataZoneClient(ctx)
+	var state dEnvProfile
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+
+	_, err := conn.DeleteEnvironmentProfile(ctx, &datazone.DeleteEnvironmentProfileInput{
+		DomainIdentifier: state.DomainIdentifier.ValueStringPointer(),
+		Identifier:       state.Id.ValueStringPointer(),
+	})
+
+	if err != nil && !errs.IsA[*awstypes.ResourceNotFoundException](err) {
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.DataZone, create.ErrActionSetting, ResNameEnvironmentProfile, state.Id.ValueString(), err),
+			err.Error(),
+		)
+		return
+	}
+}
+
+func (r *resourceEnvironmentProfile) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	parts := strings.Split(req.ID, ":")
```

Comment:
For new resources, multi-part keys should be comma-delimited. Ref: https://github.com/hashicorp/terraform-provider-aws/issues/27843